sycl/test-e2e/syclcompat/math/math_ops.cpp

```diff
@@ -0,0 +1,287 @@
+void test_syclcompat_fmin_nan() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+
+  constexpr syclcompat::dim3 grid{1};
+  constexpr syclcompat::dim3 threads{1};
+  constexpr ValueT op1 = static_cast<ValueT>(5);
+  constexpr ValueU op2 = static_cast<ValueU>(10);
+  ValueU op3 = sycl::nan(static_cast<unsigned int>(0));
+
+  constexpr std::common_type_t<ValueT, ValueU> res =
+      static_cast<std::common_type_t<ValueT, ValueU>>(5);
+
+  BinaryOpTestLauncher<ValueT, ValueU>(grid, threads)
+      .template launch_test<fmin_nan_kernel<ValueT, ValueU>>(op1, op2, res);
+
+  BinaryOpTestLauncher<ValueT, ValueU>(grid, threads)
+      .template launch_test<fmin_nan_kernel<ValueT, ValueU>>(op1, op3, op3);
+}
+
+template <typename ValueT, typename ValueU>
+inline void fmax_nan_kernel(ValueT *a, ValueU *b,
+                            std::common_type_t<ValueT, ValueU> *r) {
+  *r = syclcompat::fmax_nan(*a, *b);
+}
+
+template <typename ValueT, typename ValueU = ValueT>
+void test_syclcompat_fmax_nan() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+
+  constexpr syclcompat::dim3 grid{1};
+  constexpr syclcompat::dim3 threads{1};
+  constexpr ValueT op1 = static_cast<ValueT>(5);
+  constexpr ValueU op2 = static_cast<ValueU>(10);
+  ValueU op3 = sycl::nan(static_cast<unsigned int>(0));
+
+  constexpr std::common_type_t<ValueT, ValueU> res =
+      static_cast<std::common_type_t<ValueT, ValueU>>(10);
+
+  BinaryOpTestLauncher<ValueT, ValueU>(grid, threads)
+      .template launch_test<fmax_nan_kernel<ValueT, ValueU>>(op1, op2, res);
+
+  BinaryOpTestLauncher<ValueT, ValueU>(grid, threads)
+      .template launch_test<fmax_nan_kernel<ValueT, ValueU>>(op1, op3, op3);
+}
+
+template <typename ValueT, typename ValueU>
+inline void pow_kernel(ValueT *a, ValueU *b, ValueT *r) {
+  *r = syclcompat::pow(*a, *b);
+}
+
+template <typename ValueT, typename ValueU = ValueT>
+void test_syclcompat_pow() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+
+  // FIXME: non-floating point values default to double, requires fp64. Change
+  // when problem is solver at the header.
+  if constexpr (!std::is_floating_point_v<ValueT>) {
+    if (!syclcompat::get_current_device().has(sycl::aspect::fp64)) {
+      std::cout << "  sycl::aspect::fp64 not supported by the SYCL device."
+                << std::endl;
+      return;
+    }
+  }
+
+  constexpr syclcompat::dim3 grid{1};
+  constexpr syclcompat::dim3 threads{1};
+  // 3 ** 3 = 27
+  constexpr ValueT op1 = static_cast<ValueT>(3);
+  constexpr ValueU op2 = static_cast<ValueU>(3);
+  constexpr ValueT res = static_cast<ValueT>(27);
+
+  BinaryOpTestLauncher<ValueT, ValueU>(grid, threads)
+      .template launch_test<pow_kernel<ValueT, ValueU>>(op1, op2, res);
+}
+
+template <typename ValueT> inline void relu_kernel(ValueT *a, ValueT *r) {
+  *r = syclcompat::relu(*a);
+}
+
+template <typename ValueT> void test_syclcompat_relu() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
```

Comment:
You should sync with @hdelan on the relu implementation.

Reply:
Thanks for the pointer. This has to work in every backend though. I will sync with hdelan and see how to tackle this for each backend, as this implementation is probably quite slow.